src/main/java/utils/DriverManager.java

```diff
@@ -0,0 +1,58 @@
+package utils;
+
+import configurations.ConfigReader;
+import io.appium.java_client.AppiumDriver;
+import io.appium.java_client.android.AndroidDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.net.URI;
+import java.time.Duration;
+
+public class DriverManager {
+    private static AppiumDriver driver;
+
+    private static final String DEVICE_NAME = ConfigReader.emulatorConfig.deviceName();
+    private static final String PLATFORM_NAME = ConfigReader.emulatorConfig.platformName();
+    private static final String APP_PACKAGE = ConfigReader.emulatorConfig.appPackage();
+    private static final String APP_ACTIVITY = ConfigReader.emulatorConfig.appActivity();
+    private static final String URL = ConfigReader.emulatorConfig.remoteURL();
+    private static final String AUTOMATION_NAME = ConfigReader.emulatorConfig.automationName();
+
+    private static DriverManager INSTANCE;
+
+    public static DriverManager getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new DriverManager();
+        }
+        return INSTANCE;
+    }
+
+    public AppiumDriver initDriver() throws Exception {
```

Comment:
Please pass driver capabilities as an argument. We expect runing the tests against different devices and we should be able to reuse the methods